### DISCUSSION TITLE
x-reactions 执行 js代码时的安全问题

### DISCUSSION THREAD

**Discussion by jackchenl** [alibaba/formily#2974] [Q&A 2.x] [answered]
比如，某用户或者配置表单的人员，在x-reactions 中注入了恶意的js代码，通过 document.cookie 获取到 cookie 或其他信用凭证 发送给他自己，或者直接通过 document 乱改页面等等 恶意的行为。
请问如何解决 该类安全问题？

**Comment by janryWang** [comment, marked as answer]
首先，json-schema并不是给任何用户消费的，这是属于研发数据，不容易被篡改的，当然，前提是你不要把表达式输入权限开放给C端用户

**Reply by jackchenl** [reply]
好的，谢谢了
